modules/dicom_archive/php/viewdetails.class.inc

```diff
@@ -148,14 +148,36 @@ class ViewDetails extends \NDB_Form
         $config =& \NDB_Config::singleton();
         $dicomArchiveSettings = $config->getSetting('imaging_modules');
 
+        // escape any forward slashes
+        $pNameRegex      = preg_replace(
+            '/\//',
+            '\/',
+            $dicomArchiveSettings['patientNameRegex']
+        );
+        $legoPhanRegex   = preg_replace(
+            '/\//',
+            '\/',
+            $dicomArchiveSettings['LegoPhantomRegex']
+        );
+        $livingPhanRegex = preg_replace(
+            '/\//',
+            '\/',
+            $dicomArchiveSettings['LivingPhantomRegex']
+        );
+        $pIDRegex        = preg_replace(
+            '/\//',
+            '\/',
+            $dicomArchiveSettings['patientIDRegex']
+        );
+
```

Comment:
The `preg_match` tests below do not seem to take into account whether the archive really *is* of phantom scans or not. I agree that that's the best we can do for projects that do not use the imaging uploader but for those that do, we can look at table `mri_upload` in order to know this. This way, we can perform the test that is appropriate to the type of archive considered.